crawl-ref/source/religion.cc

```diff
@@ -2048,8 +2048,9 @@ bool do_god_gift(bool forced)
         }                       // switch (you.religion)
     }                           // End of gift giving.
 
-    if (success)
-        you.running.stop();
+    if (success && you.running.runmode != 0) // interrupt and clear runrest
```

Comment:
I think this should be using `stop_running` instead of reaching into `you.running` directly. `stop_running` also supports a `clear_delays` parameter, defaulting to true. It seems like if you set this to false instead, you might be able to call it unconditionally and avoid the `you.running.runmode != 0` check. Does that sound right to you?

Reply:
I think what you're describing will work too.

I used `you.running.stop` just because the original did, but yes, changing it to `stop_running` seems reasonable.

Calling `stop_running(false)` in `do_god_gift` fixes the bug as well. But it will also run checks (which in the case we're concerned with should never be passed) in `runrest::stop` to see if the screen should be redrawn, and to see if the user's Lua function on stopping running should be called. It's not clear to me whether it is better to check there or in `do_god_gift`.

(If the check remains in `do_god_gift` I'll change the condition to just check `success && you.running`, not `you.running.runmode != 0`; sorry that was a little goofy.)

In any case, let me know what you think and I'll push one or both changes.

Reply:
I think as long as we never actually call the user's Lua hooks unnecessarily, I'd rather the check be simple (i.e., just call `stop_running(false)`) even if this is slightly wasteful. If there's some chance we call the user's Lua hooks when we're not actually planning to stop running, then I'd prefer to avoid that even if it makes the check more complicated.

Reply:
I've pushed a new version; sorry such a small change turned into such a task.